ops/manifests/manifest.py

```diff
@@ -343,5 +343,22 @@ def delete_resources(
                     log.exception(f"{log_msg}")
                     raise ManifestClientError(log_msg) from ex
 
+    def _delete(
+        self, obj: HashableResource, namespace: Optional[str], ignore_labels: bool
+    ):
+        if ignore_labels:
+            self.client.delete(type(obj.resource), obj.name, namespace=namespace)
+        else:
+            for item in self.client.list(
+                type(obj.resource),
+                namespace=namespace,
+                labels={
+                    "juju.io/application": self.model.app.name,
+                    "juju.io/manifest": self.name,
+                },
+                fields={"metadata.name": obj.name},
+            ):
+                self.client.delete(type(item), item.metadata.name, namespace=namespace)
```

Comment:
The new behavior is to list all resources of this type, namespace, name, and matching labels, and delete that one